src/logics/ArithLogic.cc

```diff
@@ -269,18 +269,24 @@ void ArithLogic::termSort(vec<PTRef>& v) const
     sort(v, LessThan_deepPTRef(*this));
 }
 
-std::string ArithLogic::protectName(std::string const & name, SRef retSort, bool isNullary) const {
+bool ArithLogic::isAmbiguousNullarySymbolName(std::string_view name) const {
+    int i = 0;
+    auto name_str = std::string(name);
+    for (auto sr : term_store.getHomonymousNullarySymbols(name)) {
+        if (not (sym_store[sr].isInterpreted() and (opensmt::isIntString(name_str.c_str()) or opensmt::isRealString(std::string(name_str).c_str())))) {
```

Comment:
```suggestion
        if (not (sym_store[sr].isInterpreted() and (opensmt::isIntString(name_str.c_str()) or opensmt::isRealString(name_str.c_str())))) {
```
This seems like a redundant copy.

Reply:
Yes, I must have forgotten to fix the second.